python/lsst/ap/pipe/metrics.py

```diff
@@ -187,3 +189,133 @@ def run(self, matchedFakes, band):
         meas = Measurement(self.config.metricName,
                            len(magCutFakes) * u.count)
         return Struct(measurement=meas)
+
+
+class PipelineTimingMetricConnections(
+        MetricTask.ConfigClass.ConnectionsClass,
+        dimensions={"instrument", "visit", "detector"},
+        defaultTemplates={"labelStart": "",
+                          "labelEnd": "",
+                          "package": "ap_pipe",
+                          "metric": "ApPipelineTime"}):
+    metadataStart = connTypes.Input(
+        name="{labelStart}_metadata",
+        doc="The starting task's metadata.",
+        storageClass="TaskMetadata",
+        dimensions={"instrument", "exposure", "detector"},
+        multiple=False,
+    )
+    metadataEnd = connTypes.Input(
+        name="{labelEnd}_metadata",
+        doc="The final task's metadata.",
+        storageClass="TaskMetadata",
+        dimensions={"instrument", "visit", "detector"},
+        multiple=False,
+    )
+
+
+class PipelineTimingMetricConfig(MetricTask.ConfigClass, pipelineConnections=PipelineTimingMetricConnections):
+    # Don't include the dimensions hack that MetadataMetricConfig has; unlike
+    # TimingMetricTask, this task is not designed to be run on multiple
+    # pipelines.
+    targetStart = pexConfig.Field(
+        dtype=str,
+        doc="The method to profile on the starting task, optionally prefixed by "
+            "one or more tasks in the format of "
+            "`lsst.pipe.base.Task.getFullMetadata()`.")
+    targetEnd = pexConfig.Field(
+        dtype=str,
+        doc="The method to profile on the final task, optionally prefixed by "
+            "one or more tasks in the format of "
+            "`lsst.pipe.base.Task.getFullMetadata()`.")
+
+
+class PipelineTimingMetricTask(AbstractMetadataMetricTask):
+    """A Task that computes a wall-clock time for an entire pipeline, using
+    metadata produced by the `lsst.utils.timer.timeMethod` decorator.
+
+    Parameters
+    ----------
+    args
+    kwargs
+        Constructor parameters are the same as for
+        `lsst.verify.tasks.MetricTask`.
```

Comment:
I would assume that if no parameters are listed, then the constructor takes no parameters. In any case, there's no "natural" link to `MetricTask` (its great-grandparent), even though it would be useful for anyone looking at an implementation class.

Reply:
Ah, the class hierarchy wasn't fully clear to me.